src/accessories/TokenArranger.ttslua

```diff
@@ -157,6 +145,56 @@ function loadDefaultValues()
   }
 end
 
+-- creates all starting buttons and inputs
+function createDefaultButtons(loadInputs)
+  loadInputs = loadInputs or false
+
+  buttonParameters.function_owner = self
+  buttonParameters.label          = ""
+  buttonParameters.tooltip        = "Add / Remove"
+  buttonParameters.color          = { 0, 0, 0, 0 }
+  buttonParameters.width          = 325
+  buttonParameters.height         = 325
+  -- create UI
+  local offset = 0.725
+  local pos = { x = { -1.067, 0.377 }, z = -2.175 }
+
+  -- button and inputs index 0-9
+  for i = 1, 10 do
+    if i < 6 then
+      buttonParameters.position = { pos.x[1], 0, pos.z + i * offset }
+      if loadInputs then
+        inputParameters.position  = { pos.x[1] + offset, 0.1, pos.z + i * offset }
+      end  
+    else
+      buttonParameters.position = { pos.x[2], 0, pos.z + (i - 5) * offset }
+      if loadInputs then
+        inputParameters.position  = { pos.x[2] + offset, 0.1, pos.z + (i - 5) * offset }
+      end     
+    end
+
+    buttonParameters.click_function = attachIndex("tokenClick", i)
+    self.createButton(buttonParameters)
+
+    if loadInputs then
+      inputParameters.input_function  = attachIndex2("tokenInput", i)
+      inputParameters.value           = tokenPrecedence[TOKEN_NAMES[i]][1]
+      self.createInput(inputParameters)
+    end
```

Comment:
What's the use of this? If I understand it correctly, this optionally removes the input fields.

Reply:
This is so that the button creation can all be one function, but so the input fields are only created onLoad.  

This is because the way the script handles it is that all buttons are deleted whenever the copied tokens are deleted (since the percentage text is on buttons) and then the core buttons are immediately recreated.  This is to avoid having to deal with button indexing, which can get a bit hairy in my experience (so I just use the clearButtons() method).  No reason to do that to the input fields though, so they only need to be created once; however, since they depend on similar local variables, I saw fit to put them in the function as well.  I can try to carve this bit out and put it back in onLoad if you think that would help clarity though.

Reply:
Ah, now I see what this is about. Since the buttons are cleared when updating the percentages, we only need to recreate the buttons and not the input field.
I think it would be a bit clearer if the function name indicated this (maybe `createButtonsAndInputs()`?). Could also split the input field creation into its own function (and only call that in onLoad() and not at the later point). Alternatively, you could also work with `editButton()` instead of removing all of them (though this is probably a bit cumbersome).

Reply:
Yeah, editButton() I want to avoid because button indexing is a bit of a mess in my experience. I've decided to stick to renaming the function for added clarity.  Will update once commit is pushed

Reply:
Done!